chains/solana/contracts/programs/example-ccip-receiver/src/lib.rs

```diff
@@ -351,6 +372,8 @@ pub enum CcipReceiverError {
     OnlyOwner,
     #[msg("Address is not proposed_owner")]
     OnlyProposedOwner,
+    #[msg("Offramp is not allowed")]
+    OfframpNotAllowed,
```

Comment:
Wdyt about renaming the error to a simple `Unauthorized` or `InvalidCaller`? The way this is phrased right now assumes that the caller is in fact an offramp, and that it just happens to not be allowed. But it could be the case that the caller is just some random (or malicious) user/program and not even an offramp, so something more generic would be appropriate then. 
I've checked and [EVM token pools](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/ccip/pools/TokenPool.sol#L605) (which do a similar auth flow) define an `CallerIsNotARampOnRouter` error, which feels verbose to me but would also be a possibility (note: token pools get called by both onramp and offramp, unlike the receiver).